website/docs/advanced-concepts/access-control/granular-access-control/roles.md

```diff
@@ -0,0 +1,242 @@
+**Edit** allows the user to change/modify the existing datasources. You can edit the datasource name, configurations and further test and save the datasource. Edit permission can be given at different hierarchical levels -
+
+##### Workspace
+
+If you give edit permission to the workspace, it gives access to modify all the datasources in the respective workspace. 
+
+##### Datasource
+
+If you want the user to have edit access only to a specific datasource in a workspace, open the respective workspace tree and check the edit permission adjacent to the respective datasource. **Edit** for just a datasource allows you to modify the name, configurations of that datasource.
+
+#### View
+
+**View** only allows a user to have view access for the existing datasources. The user can interact with the functionality of the datasource, test the datasource, but can't make any changes with View permission. View permission can be given at different hierarchical levels -
+
+##### Workspace
+
+If you give view permission to the workspace, it gives access to view and execute all the datasources in the respective workspace.
+
+##### Datasource 
+
+If you want the user to have view access only to a specific datasource in a workspace, open the respective workspace tree and check the view permission adjacent to the respective datasource. **View** for just a datasource allows you to view, test and execute that datasource.
+
+#### Delete
+
+**Delete** gives the user an access to delete the existing datasources in a workspace. With the delete permission, one can't edit or create a datasource, but, the user has the access to delete it.
+Delete permission can be given at various hierarchical levels - 
+
+##### Workspace
+
+If you give delete permission to the workspace, It gives access to remove any datasource from the respective workspace. 
+
+##### Datasource
+
+**Delete** for a particular datasource allows you to remove that datasource only.
+
+### Groups and roles permissions
+
+For Groups and roles for an Appsmith instance, here are the permissions that can be assigned to user or a group -
+
+#### Create
+
+With this permission, users can create new groups and roles. 
+
+##### Groups
+
+When the create permission is enabled for groups, a user can create a new group. This also automatically enables permission to view, edit, delete, invite, or remove a user from an existing group.
+
+##### Roles
+
+With create permission enabled for roles, a user can create a new role. This also automatically enables permission to view, edit, delete, the existing roles.
+
+#### Edit 
+
+With this permission, users can edit the existing groups and roles. 
+
+##### Groups
+
+When the edit permission is enabled for groups, a user can edit an existing group. 
+
+##### Roles
+
+With edit permission enables for roles, a user can modify an existing role.
+
+:::info
+
+You can't edit the built-in roles from Appsmith. You can only the edit the custom roles defined by any user in your instance.
```

Comment:
There is a `the` before edit... it's not required, the statement isn't correct.

Reply:
Strange that linter didn't pick this. cc @jnikhila

Reply:
@Pranay105 - linter configuration that we have only ensures consistency. We still need to run it through grammarly for grammar check. However, we can extend vale to do grammar checks for us. We can evaluate and co figure that. Please add a separate ticket to look into it.